src/ManualClient.py

```diff
@@ -511,18 +511,16 @@ def build_tracker_and_locations_table(self):
                         self.listed_locations["(No Category)"].append(location_id)
 
                 victory_location =  self.ctx.goal_location
-                victory_categories = set()
+                victory_categories = set(victory_location.get("category", []))
 
-                if "category" in victory_location and len(victory_location["category"]) > 0:
-                    for category in victory_location["category"]:
-                        if category not in self.location_categories:
-                            self.location_categories.append(category)
+                for category in victory_categories:
+                    if category not in self.location_categories:
+                        self.location_categories.append(category)
 
-                        if category not in self.listed_locations:
-                            self.listed_locations[category] = []
-                            victory_categories.add(category)
-
-                if not victory_categories:
+                    if category not in self.listed_locations:
+                        self.listed_locations[category] = []
+                
+                if len(victory_categories) == 0:
```

Comment:
Is there a reason you're using `len() == 0` instead of `not`?

Reply:
Stemmed from a misunderstanding of how for-else works in Python, which I originally did for the for loop above. In my head, for-else would do the for if there are any iterations and do the else if no iterations. In actuality, Python just like... does both if you don't break the loop. Which is kinda goofy, lol

I can change it to not :smile: